plugins/miopen_legacy_plugin/engines/plans/miopen_batchnorm_fwd_inference_plan.cpp

```
@@ -0,0 +1,163 @@
+{
+    return *_x_pair;
+}
+
+const Miopen_tensor& Batchnorm_fwd_inference_params::y() const
+{
+    return *_y_pair;
+}
+
+const Miopen_tensor& Batchnorm_fwd_inference_params::scale() const
+{
+    return *_scale_pair;
+}
+
+const Miopen_tensor& Batchnorm_fwd_inference_params::bias() const
+{
+    return *_bias_pair;
+}
+
+const std::optional<std::unique_ptr<Miopen_tensor>>&
+    Batchnorm_fwd_inference_params::est_mean() const
+{
+    return _est_mean_tensor_descriptor;
+}
+
+const std::optional<std::unique_ptr<Miopen_tensor>>&
+    Batchnorm_fwd_inference_params::est_variance() const
+{
+    return _est_variance_tensor_descriptor;
+}
+
+void Batchnorm_fwd_inference_params::initialize_tensors(
+    const hipdnn_sdk::data_objects::BatchnormInferenceAttributes& attributes,
+    const std::unordered_map<int64_t, const hipdnn_sdk::data_objects::TensorAttributes*>&
+        tensor_map)
+{
+    if(auto x_tensor_attr = tensor_map.find(attributes.x()); x_tensor_attr != tensor_map.end())
+    {
+        _x_pair = std::make_unique<Miopen_tensor>(*x_tensor_attr->second);
+    }
+    if(auto y_tensor_attr = tensor_map.find(attributes.y()); y_tensor_attr != tensor_map.end())
+    {
+        _y_pair = std::make_unique<Miopen_tensor>(*y_tensor_attr->second);
+    }
+    if(auto scale_tensor_attr = tensor_map.find(attributes.scale());
+       scale_tensor_attr != tensor_map.end())
+    {
+        _scale_pair = std::make_unique<Miopen_tensor>(*scale_tensor_attr->second);
+    }
+    if(auto bias_tensor_attr = tensor_map.find(attributes.bias());
+       bias_tensor_attr != tensor_map.end())
+    {
+        _bias_pair = std::make_unique<Miopen_tensor>(*bias_tensor_attr->second);
+    }
+
+    if(attributes.mean().has_value())
+    {
+        if(auto est_mean_attr = tensor_map.find(attributes.mean().value());
+           est_mean_attr != tensor_map.end())
+        {
+            _est_mean_tensor_descriptor = std::make_unique<Miopen_tensor>(*est_mean_attr->second);
+        }
```

Should throw, or log if it's missing and the attributes expects it I think.